go/libraries/doltcore/sqle/enginetest/dolt_queries_diff.go

```diff
@@ -528,14 +528,44 @@ var DiffSystemTableScriptTests = []queries.ScriptTest{
 			},
 			{
 				Query:    "SELECT COUNT(*) FROM DOLT_DIFF_t;",
-				Expected: []sql.Row{{1}},
+				Expected: []sql.Row{{7}},
 			},
 			{
 				Query:    "SELECT to_pk, to_c1, from_pk, from_c1, diff_type FROM DOLT_DIFF_t where to_commit=@Commit4;",
 				Expected: []sql.Row{{7, 8, nil, nil, "added"}},
 			},
 		},
 	},
+	{
+		// Similar to previous test, but with one row to avoid ordering issues.
+		Name: "altered keyless table add pk with removed", // https://github.com/dolthub/dolt/issues/8625
+		SetUpScript: []string{
+			"create table tbl (i int, j int);",
+			"insert into tbl values (42, 23);",
+			"call dolt_commit('-Am', 'commit1');",
+			"alter table tbl add primary key(i);",
+			"call dolt_commit('-am', 'commit2');",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query: "SELECT to_i,to_j,from_i,from_j,diff_type  FROM dolt_diff_tbl;",
+				// Output in the situation is admittedly wonky. Updating the PK leaves in a place where we can't really render
+				// the diff, but we want to show something. In this case, the 'pk' column tag changes, so in the last two rows
```

Comment:
Adding the primary key changes the tags of the pk columns (i.e. `i`)? I thought that only happened on type (family) changes from altering the column directly. 

I think I was expecting that if we didn't have a tag match, we'd fall back to a name match, and then we'd still be able to correctly match up `i`. I see now that we're using `MapSchemaBasedOnTagAndName` but only in the `IsDiffablePartition` function to return a boolean. If we were able to use `MapSchemaBasedOnTagAndName` for the mapping in this case, too, would that cause the correct value of `i` to show up in the diff table? 

I like that we're showing more data here; it's just a little confusing, like your comment mentions, since the data isn't a real value of the `i` column that was inserted, so seems a little misleading to read it.